invenio_files_rest/tasks.py

```diff
@@ -279,3 +281,30 @@ def remove_expired_multipartobjects():
 
     for fid in file_ids:
         remove_file_data.delay(fid)
+
+
+@shared_task(ignore_result=True)
+def clear_orphaned_files():
+    """Delete orphaned files from DB and storage.
+
+    .. note:: Orphan files are files
+        (:class:`invenio_files_rest.models.FileInstance` objects and their
+        on-disk counterparts) that do not have any
+        :class:`invenio_files_rest.models.ObjectVersion` objects associated
+        with them (anymore).
+
+    :param force: Whether to delete the file even if the file instance is not
+        marked as writable.
+    """
+    # with the tilde (~) operator, we get all file instances that *don't*
+    # have any associated object versions
+    orphans = FileInstance.query.filter(~FileInstance.objects.any()).all()
```

Comment:
This query might return high number of results, if not often run or the first time? Or do you want it will always have low cardinality?
It might create high number of celery task.

Reply:
Good point, I haven't taken that into consideration (because that's likely not an issue for us any time soon).
Maybe we should add a (configurable) constraint for the maximum number of handled files per execution, with a default value of about 1000 or so?

The only case in which this wouldn't necessarily "work" would be if more orphan files are added between executions than the configured limit... But in that case, one could just increase the limit or schedule the task more frequently.

Reply:
Oh, and of course we could handle a configured limit of `0` to mean "no limit", as is often done... That gives the option to choose freely.

Reply:
What about having another param, with a default value, that you can configure when you add the celery task in the config as above?
```
'clear-orphan-files': {
  'task': 'invenio_files_rest.tasks.clear_orphaned_files',
  'schedule': 60 * 60 * 24,
  'kwargs': {
    'force': True,
    'limit': 100
  }
},
```
and then:
```suggestion
    orphans = FileInstance.query.filter(~FileInstance.objects.any()).limit(limit).all()
```
?

Reply:
that's pretty much what i was thinking :)
i'll update the PR soon!